File: backend/features/highlights/highlights-core/src/main/java/de/bennyboer/kicherkrabbe/highlights/Link.java
package de.bennyboer.kicherkrabbe.highlights;

import lombok.AllArgsConstructor;
import lombok.Value;

import static de.bennyboer.kicherkrabbe.commons.Preconditions.notNull;
import static lombok.AccessLevel.PRIVATE;

@Value
@AllArgsConstructor(access = PRIVATE)
public class Link {

    LinkType type;

    LinkId id;

    LinkName name;

    public static Link of(LinkType type, LinkId id, LinkName name) {
        notNull(type, "Type must be given");
        notNull(id, "Id must be given");
        notNull(name, "Name must be given");

        return new Link(type, id, name);
    }

    @Override
    public String toString() {
        return "Link(type=%s, id=%s, name=%s)".formatted(type, id, name);
    }

}

File: frontend/management/src/app/modules/admin/modules/highlights/model/link.ts
import { validateProps } from '@kicherkrabbe/shared';
import { LinkType } from './link-type';

export class Link {
  readonly type: LinkType;
  readonly id: string;
  readonly name: string;

  private constructor(props: { type: LinkType; id: string; name: string }) {
    validateProps(props);

    this.type = props.type;
    this.id = props.id;
    this.name = props.name;
  }

  static of(props: { type: LinkType; id: string; name: string }): Link {
    return new Link(props);
  }
}
